server.js

```diff
@@ -26,10 +50,83 @@ app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.status(200).json({
+    Hello: "Here you can see all my routes!",
+    Routes: [
+      { "/songs": "All the song data" },
+      { "/songs/id/:id": "A specific song, for instance use id 637f5d218aae5dd90801358b" },
+      { "/songs/?danceability=70" : "or instead of 70 try another pase for instace 50"},
+      { "/songs/?genre=pop" : "or instead of pop try another genre, for instance trap music"},
+    ],
+  });
 });
 
+// app.get("/songs", async (req, res) => {
+//   const allTheSongs = await Song.find({}); //Song is the model variable
+//   res.status(200).json({
+//     success: true,
+//     body: allTheSongs
+//   });
+// });
+
+
+//get a single song by using findById
+app.get("/songs/id/:id", async (req, res) => {
+  try {
+    const singleSong = await Song.findById(req.params.id);
+    if (singleSong) {
+      res.status(200).json({
+        success: true,
+        body: singleSong
+      });
+    } else {
+      res.status(404).json({
+        success: false,
+        body: {
+        message: "Could not find the song"
+        }
+      });
+    }
+  } catch(error) {
+    res.status(400).json({
+      success: false,
+      body: {
+      message: "Invalid id"
+      }
+    });
+  }
+});
+
+app.get("/songs/", async (req, res) => {
```

Comment:
Songs and genre does not work for us. We get this error message: Cast to Number failed for value \"/.*/\" (type RegExp) at path \"danceability\" for model \"Song\""